internal/services/dataprotection/data_protection_backup_policy_postgresql_resource_test.go

```diff
@@ -13,27 +13,39 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
 	"github.com/hashicorp/terraform-provider-azurerm/utils"
 )
 
 type DataProtectionBackupPolicyPostgreSQLResource struct{}
 
-func TestAccDataProtectionBackupPolicyPostgreSQL_basic(t *testing.T) {
+func TestAccDataProtectionBackupPolicyPostgreSQL_basicDeprecatedInV4(t *testing.T) {
+	if features.FourPointOhBeta() {
+		t.Skip("this test requires 3.0 mode")
+	}
 	data := acceptance.BuildTestData(t, "azurerm_data_protection_backup_policy_postgresql", "test")
 	r := DataProtectionBackupPolicyPostgreSQLResource{}
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
-			Config: r.basic(data),
+			Config: r.basicDeprecatedInV4(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 			),
 		},
-		data.ImportStep(),
+		data.ImportStep(
+			"default_retention_duration",
+			"default_retention_rule.#",
+			"default_retention_rule.0.%",
+			"default_retention_rule.0.life_cycle.#",
+			"default_retention_rule.0.life_cycle.0.%",
+			"default_retention_rule.0.life_cycle.0.data_store_type",
+			"default_retention_rule.0.life_cycle.0.duration",
```

Comment:
are these not returned from the API? how come we are ignoring them?

Reply:
The API returns their values. However, TF supports both `default_retention_duration ` (deprecated in v4.0) and `default_retention_rule`, which are two mutually exclusive properties in v3.0. These two properties corresponding to the same property of the API. Therefore, when importing resource in v3.0, I assume that we have no way of knowing whether the `default_retention_duration ` or `default_retention_rule` is  specified in the config,  so I ignore them. Please see the following code in resourceDataProtectionBackupPolicyPostgreSQLRead func for details. Other than that, could you please let me know if there is a better solution?
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/39109137/e347c501-f964-43c5-98bc-c08c808c57cd)

Reply:
Hi @katbyte thank you very much, I think this is a very good way. But there is a little problem. If the deprecated and replaced properties are set in read at the same time, then the following checks need to be deleted at the same time. Do you think it's okay to remove the check?
![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/39109137/71a4c050-2724-4184-8773-ac891943ed40)

Reply:
Hi @katbyte a kind reminder in case you miss the above message.